feature-selection/feature-selection-main.metta

```diff
@@ -0,0 +1,71 @@
+                ($ft-set-pop-size (size-atom $ft-set-pop))                     
+                ($capped-ft-set-pop (if (and (> $diversity-cap 0) (< $diversity-cap $ft-set-pop-size))                  ;; limmit the size of feature set pop to diversity cap value
+                                        (takeN $diversity-cap $ft-set-pop)
+                                        $ft-set-pop))
+                (() (println! (Selected features set pop $capped-ft-set-pop))))
+                (selectTopFts (mkITable $table $labels) $div-pres -1 $multi-diversity $sim-scorer $capped-ft-set-pop $nDeme)))                                                              
+                                                                                                                                
+(= (featureSelection            ;; for sake of clarity I chose to list the parameteres top-down style
+            $algo
+            (mkITable $table $labels)
+            $th                 ;; threshold for simle, smd ..
+            $target-size        ;; desired number of features in a set   
+            $init-fts           ;; indices of initial features -- ne
+            ) 
+	(case $algo
+		((hc (let $init-instance (List.generate (- (List.length $labels) 1) 0) 
+                        (hcFtSetPop (mkITable $table $labels) $init-instance)))
+		 (rd (randomFeatureSelection (mkITable $table $labels) $target-size))                                       
+		 (sim (simpleFeatureSelector $th $target-size False (mkITable $table $labels) ()))                           ;; False -- default value for using exp distribution for selection of top N features -- and start with empty feature list as initial features -- th here refers to the minimum score required for feature set to be selected
+		 (inc (incrementalSelection (mkITable $table $labels) $th $target-size 1 0))                    ;; 1 -- default value of max-interaction-terms -- the maximum size of each feature set tested in the scorer -- the last value is a value redundancy threshold values
+		 (smd (if (== $init-fts ())
+                        (stochasticMaxDependency $th 3 $target-size (mkITable $table $labels) mi ())
+                        (let* (($len (List.length $labels))
+                                ($oc (Table.getColumn (- $len 1) $table))      ;; output column to calculate MI for initial ft set
+                                ($init-ft-zip (zipColumn $init-fts $table))     ;; 
+                                ($mi (mutualInformation $init-ft-zip $oc)))
+                                
+                                (stochasticMaxDependency $th $target-size 3 (mkITable $table $labels) mi (($mi $init-fts))))))          ;; this `th` is a vlue used to check marginal score improvements as a features are added to the set. feature-count refers to the number of features in a single feature set
```

Comment:
The way I see it, the `$target-size` and `3` are swapped in position here and in line number 63. Was that intentional?